### PR TITLE
Fix creating dummy context for subscription webhooks.

### PR DESCRIPTION
**Pull request by SzymJ**
I want to merge this change because it fixes creating context used in subscription webhooks.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6981 Webhook Create with custom query is not working](https://app.clickup.com/t/2549495/SALEOR-6981)

File: saleor/graphql/webhook/subscription_payload.py [hunk 1]
```diff
@@ -54,10 +55,11 @@ def initialize_context() -> HttpRequest:
     return: HttpRequest
     """
     handler = BaseHandler()
-    context = RequestFactory().request()
+    context = RequestFactory().request(**{"SERVER_NAME": settings.ALLOWED_HOSTS[-1]})
```
Comment (maarcingebala) [inline saleor/graphql/webhook/subscription_payload.py:58]: Why -1? See how we [set up this value for WSGI app](https://github.com/saleor/saleor/blob/ebc0e0e66168c358373d594c0c50865ef8e6507f/saleor/wsgi/__init__.py#L37), maybe we should reuse it here.

File: saleor/graphql/webhook/subscription_payload.py [hunk 2]
```diff
@@ -54,10 +56,13 @@ def initialize_context() -> HttpRequest:
     return: HttpRequest
     """
     handler = BaseHandler()
-    context = RequestFactory().request()
+    context = RequestFactory().request(
+        **{"SERVER_NAME": SimpleLazyObject(lambda: settings.ALLOWED_HOSTS[0])}
```
Comment (korycins) [inline saleor/graphql/webhook/subscription_payload.py:60]: Just to confirm. Are you sure, that this is OK to provide only the first value of allowed hosts?
Why do we need a SimpleLazyObject? Is it requirement for request?

Comment (patrys) [reply saleor/graphql/webhook/subscription_payload.py:60]: Also why create a dict to immediately unpack it to keyword args?

Comment (SzymJ) [reply saleor/graphql/webhook/subscription_payload.py:60]: I think that in this case there is no matter which element of `ALLOWED_HOSTS` we choose, @maarcingebala suggested in the same way as in[ WSGI app](https://github.com/saleor/saleor/blob/ebc0e0e66168c358373d594c0c50865ef8e6507f/saleor/wsgi/__init__.py#L37) so I did that.

Comment (maarcingebala) [reply saleor/graphql/webhook/subscription_payload.py:60]: @patrys Didn't we want to stop relying on domains from site settings and deprecate/drop the API to set it?

Comment (maarcingebala) [reply saleor/graphql/webhook/subscription_payload.py]: OK so let's use site_settings then.

Comment (SzymJ) [reply saleor/graphql/webhook/subscription_payload.py]: Are domains from Site objects always added to ALLOWED_HOSTS?

**Review by maarcingebala — APPROVED** [review]

File: saleor/graphql/webhook/subscription_payload.py (hunk 2 above)
Comment (patrys) [reply saleor/graphql/webhook/subscription_payload.py]: They better be or Saleor won't work 😄

**Review by IKarbowiak — APPROVED** [review]

**Pull request merged**